packages/realm-react/src/RealmContext.ts

```diff
@@ -0,0 +1,109 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2024 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { createUseObject } from "./useObject";
+import { createUseQuery } from "./useQuery";
+import { createUseRealm } from "./useRealm";
+import { FlexibleRealmProviderFC } from "./RealmProvider";
+
+export type RealmContext<RealmProvider = FlexibleRealmProviderFC> = {
+  /**
+   * The Provider component that is required to wrap any component using
+   * the Realm hooks.
+   * @example
+   * ```
+   * const AppRoot = () => {
+   *   const syncConfig = {
+   *     flexible: true,
+   *     user: currentUser
+   *   };
+   *
+   *   return (
+   *     <RealmProvider schema={[Task, User]} path={"data.realm"} sync={syncConfig}>
+   *       <App/>
+   *     </RealmProvider>
+   *   )
+   * }
+   * ```
+   * @param props - The {@link Realm.Configuration} for this Realm defaults to
+   * the config passed to `createRealmProvider`, but individual config keys can
```

Comment:
The original text said passed to `createRealmProvider` but this is a private method isn't it? `createRealmContext` seems more appropriate.